core/common/assets/js/api/core/data.js

```diff
@@ -0,0 +1,476 @@
+				return 'GET';
+
+			case 'update':
+				return 'PUT';
+		}
+
+		return false;
+	}
+
+	/**
+	 * Function getAllowedMethods().
+	 *
+	 * Returns allowed HTTP methods by type.
+	 *
+	 * @param {DataTypes} type
+	 *
+	 * @return {[string]|boolean}
+	 */
+	getAllowedMethods( type ) {
+		switch ( type ) {
+			case 'create':
+				return CREATABLE;
+
+			case 'delete':
+				return DELETABLE;
+
+			case 'get':
+				return READABLE;
+
+			case 'update':
+				return EDITABLE;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Function commandToEndpoint().
+	 *
+	 * Convert command to endpoint.
+	 *
+	 * For example `component/command/:arg_example` => `controller/endpoint/8`.
+	 *
+	 * TODO: Find a better solution.
+	 *
+	 * @param {string} command
+	 * @param {{}} args
+	 * @param {string|null} [format]
+	 *
+	 * @returns {string} endpoint
+	 */
+	commandToEndpoint( command, args, format = null ) {
+		let endpoint = command;
+
+		if ( args.query ) {
+			if ( ! format ) {
+				format = command;
+			}
+
+			if ( format && format.includes( '/{' ) ) {
+				// Means command includes magic query arguments ( controller/endpoint/{whatever} ).
+				const magicParams = format.split( '/' ).filter( ( str ) => '{' === str.charAt( 0 ) );
+
+				magicParams.forEach( ( param ) => {
+					// Remove the '{', '}'.
+					param = param.replace( '{', '' );
+					param = param.replace( '}', '' );
+
+					const formatted = Object.entries( args.query ).find( ( [ key ] ) => key === param );
+
+					if ( ! formatted ) {
+						return;
+					}
+
+					const key = formatted[ 0 ],
+						value = formatted[ 1 ].toString();
+
+					// Replace magic params with values.
+					format = format.replace( new RegExp( '{' + param + '}', 'g' ), value );
+
+					delete args.query[ key ];
+				} );
+
+				endpoint = format;
+			}
+
+			if ( endpoint.includes( 'index' ) ) {
+				endpoint = endpoint.replace( '/index', '' );
+			}
+
+			if ( args.query.id ) {
+				endpoint += '/' + args.query.id.toString();
+				delete args.query.id;
+			}
+
+			// Sorting since the endpoint later will be used as key to store the cache.
+			const queryEntries = Object.entries( args.query ).sort(
+				( [ aKey ], [ bKey ] ) => aKey - bKey // Sort by param name.
+			);
+
+			// `args.query` will become a part of GET params.
+			if ( queryEntries.length ) {
+				endpoint += '?';
+
+				queryEntries.forEach( ( [ name, value ] ) => {
+					endpoint += name + '=' + value + '&';
+				} );
+			}
+
+			// If last character is '&' remove it.
+			endpoint = endpoint.replace( /&$/, '' );
+		}
+
+		// If requested magic param does not exist in args, need to remove it to have fixed endpoint.
+		// eg: 'documents/{documentId}/elements/{elementId}' and args { documentId: 4123 }.
+		// result: 'documents/4123/elements'
+		if ( endpoint.includes( '/{' ) ) {
+			endpoint = endpoint.substring( 0, endpoint.indexOf( '/{' ) );
+		}
+
+		return endpoint;
+	}
+
+	/**
+	 * Function endpointToCommand().
+	 *
+	 * Convert endpoint to command.
```

Comment:
Example.

Reply:
This method does not exist anymore the 'alternative' for this method is: https://github.com/elementor/elementor/pull/11388/commits/cd9bc3da5a29fdf323158b3c81bc47e49d1c40e1#diff-d0a28344df7ce0852c5a7c797db34c66R196